Sources/Controllers/2.0/OASelectFavoriteGroupViewController.mm

```diff
@@ -0,0 +1,224 @@
+//
+//  OASelectFavoriteGroupViewController.m
+//  OsmAnd Maps
+//
+//  Created by nnngrach on 16.03.2021.
+//  Copyright © 2021 OsmAnd. All rights reserved.
+//
+
+#import "OASelectFavoriteGroupViewController.h"
+#import "OAFavoriteItem.h"
+#import "OAFavoritesHelper.h"
+#import "OAColors.h"
+#import "Localization.h"
+#import "OAUtilities.h"
+#import "OASettingsTableViewCell.h"
+#import "OATitleRightIconCell.h"
+#import "OAMultiIconTextDescCell.h"
+#import "OAAddFavoriteGroupViewController.h"
+#import "OsmAndApp.h"
+
+#define kCellTypeAction @"OATitleRightIconCell"
+#define kMultiIconTextDescCell @"OAMultiIconTextDescCell"
+#define kAddNewGroupSection 0
+#define kGroupsListSection 1
+
+@interface OASelectFavoriteGroupViewController() <UITableViewDelegate, UITableViewDataSource, OAAddFavoriteGroupDelegate>
+
+@end
+
+@implementation OASelectFavoriteGroupViewController
+{
+    OsmAndAppInstance _app;
+    NSString *_selectedGroupName;
+    NSArray<OAFavoriteGroup *> *_groupedFavorites;
+    NSArray<NSArray<NSDictionary *> *> *_data;
+}
+
+- (instancetype) initWithSelectedGroupName:(NSString *)selectedGroupName;
+{
+    self = [super initWithNibName:@"OABaseTableViewController" bundle:nil];
+    if (self)
+    {
+        _selectedGroupName = selectedGroupName;
+        _app = [OsmAndApp instance];
+        [self reloadData];
+        [self generateData];
+    }
+    return self;
+}
+
+- (void) viewDidLoad
+{
+    [super viewDidLoad];
+    self.tableView.delegate = self;
+    self.tableView.dataSource = self;
+    self.tableView.separatorColor = UIColorFromRGB(color_tint_gray);
+    self.tableView.tableHeaderView = [OAUtilities setupTableHeaderViewWithText:OALocalizedString(@"select_gropu_descr") font:[UIFont systemFontOfSize:15.] textColor:UIColor.blackColor lineSpacing:0. isTitle:NO];
+}
+
+- (void) applyLocalization
+{
+    [super applyLocalization];
+    self.titleLabel.text = OALocalizedString(@"select_group");
+}
+
+- (void) reloadData
+{
+    _groupedFavorites = [OAFavoritesHelper getFavoriteGroups];
+}
+
+- (void) generateData
+{
+    NSMutableArray *data = [NSMutableArray new];
+    [data addObject:@[
+        @{
+            @"type" : kCellTypeAction,
+            @"title" : OALocalizedString(@"fav_add_new_group"),
+            @"img" : @"ic_custom_add",
+        },
+    ]];
+    
+    NSMutableArray *cellFoldersData = [NSMutableArray new];
+    for (OAFavoriteGroup *group in _groupedFavorites)
+    {
+        NSString *name = group.name;
+        if (!name || name.length == 0)
+            name = OALocalizedString(@"favorites");
+        
+        [cellFoldersData addObject:@{
+            @"type" : kMultiIconTextDescCell,
+            @"header" : OALocalizedString(@"available_groups"),
+            @"title" : name,
+            @"description" : [NSString stringWithFormat:@"%i", group.points.count],
```

Comment:
Please do not ignore warnings in your code: Values of type 'NSUInteger' should not be used as format arguments; add an explicit cast to 'unsigned long' instead.
Change to %ld